modules/library/referencing/src/main/java/org/geotools/referencing/factory/IdentifiedObjectFinder.java

```diff
@@ -361,6 +368,10 @@ final IdentifiedObject createFromCodes(final IdentifiedObject object) throws Fac
         }
         return null;
     }
+    
+    protected Set getSpecificCodeCandidates(final IdentifiedObject object) throws FactoryException {
```

Comment:
No, it's a new method. The old one end up providing all code candidates (the whole set), so it's suitable only for the full scan, the new one actually makes an effort to returns only the ones relevant to the object being searched.